### PR TITLE
docs: add retriever docs

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: docs/guides/retriever/embedding-finetuning.md [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+### Sample Training Output
+
+```
+INFO:root:step 10 | epoch 0 | loss 2.1543 | grad_norm 1.2345 | lr 5.00e-06 | mem 24.50 GiB | time 3.21s
+INFO:root:step 20 | epoch 0 | loss 1.3210 | grad_norm 0.8765 | lr 5.00e-06 | mem 24.50 GiB | time 3.18s
+INFO:root:step 30 | epoch 0 | val_loss 1.1230 | val_acc1 0.7820 | val_mrr 0.8450
+```
+
+## Using the Fine-Tuned Model
```
Comment (rnyak) [inline docs/guides/retriever/embedding-finetuning.md:307]: better to change this to transformers usage script. as in here:

https://huggingface.co/nvidia/llama-nemotron-embed-1b-v2#transformers-usage

File: docs/guides/retriever/embedding-finetuning.md [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+tokenizer:
+  _target_: nemo_automodel.NeMoAutoTokenizer.from_pretrained
+  pretrained_model_name_or_path: meta-llama/Llama-3.2-1B
+
+dataloader:
+  _target_: torchdata.stateful_dataloader.StatefulDataLoader
+  dataset:
+    _target_: nemo_automodel.components.datasets.llm.make_retrieval_dataset
+    data_dir_list:
+      - /path/to/train_data.json
```
Comment (rnyak) [inline docs/guides/retriever/embedding-finetuning.md:370]: here better to use the predefined path that we added lately to automatically download the example dataset from HF hub.

see the latest yaml file pls: https://github.com/NVIDIA-NeMo/Automodel/blob/main/examples/biencoder/llama3_2_1b_biencoder.yaml#L55

File: docs/guides/retriever/embedding-finetuning.md [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+
+## Recipe Configuration
+
+The biencoder recipe is configured through a YAML file. Below is the full reference config with explanations.
+
+### Model
+
+```yaml
+model:
+  _target_: nemo_automodel.NeMoAutoModelBiencoder.from_pretrained
```
Comment (rnyak) [inline docs/guides/retriever/embedding-finetuning.md:48]: let's not merge this MR yet, we can do once MR [1449](https://github.com/NVIDIA-NeMo/Automodel/pull/1449) is finalized, we might need to update class names etc.

File: docs/guides/retriever/embedding-finetuning.md [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+## Quickstart
+
+Fine-tune a Llama 3.2 1B embedding model with a single command:
+
+```bash
+torchrun --nproc-per-node=8 examples/biencoder/finetune.py \
+    --config examples/biencoder/llama3_2_1b_biencoder.yaml
+```
+
+Or using the AutoModel CLI:
```
Comment (jgerh) [inline docs/guides/retriever/embedding-finetuning.md]:
```suggestion
Or use the AutoModel CLI:
```

File: docs/guides/retriever/embedding-finetuning.md [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+```yaml
+distributed:
+  strategy: fsdp2
+  dp_size: none
+  tp_size: 1
+  cp_size: 1
+  sequence_parallel: false
+```
+
+FSDP2 for data parallelism. Tensor parallelism available for larger backbones.
```
Comment (jgerh) [inline docs/guides/retriever/embedding-finetuning.md]:
```suggestion
FSDP2 handles data parallelism. Tensor parallelism is available for larger backbones.
```

File: docs/guides/retriever/embedding-finetuning.md [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+
+```yaml
+checkpoint:
+  enabled: true
+  checkpoint_dir: ./output/llama3_2_1b_biencoder/checkpoints
+  model_save_format: safetensors
+  save_consolidated: true
+```
+
+Saves Hugging Face `safetensors` format, directly loadable for inference or further fine-tuning.
```
Comment (jgerh) [inline docs/guides/retriever/embedding-finetuning.md]:
```suggestion
Checkpoints are saved in Hugging Face `safetensors` format and are directly loadable for inference or further fine-tuning.
```

File: docs/guides/retriever/embedding-finetuning.md [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+
+```
+INFO:root:step 10 | epoch 0 | loss 2.1543 | grad_norm 1.2345 | lr 5.00e-06 | mem 24.50 GiB | time 3.21s
+INFO:root:step 20 | epoch 0 | loss 1.3210 | grad_norm 0.8765 | lr 5.00e-06 | mem 24.50 GiB | time 3.18s
+INFO:root:step 30 | epoch 0 | val_loss 1.1230 | val_acc1 0.7820 | val_mrr 0.8450
+```
+
+## Using the Fine-Tuned Model
+
+Consolidated checkpoints are in Hugging Face format and loadable directly:
```
Comment (jgerh) [inline docs/guides/retriever/embedding-finetuning.md]:
```suggestion
Consolidated checkpoints are in Hugging Face format and are directly loadable:
```

File: docs/guides/overview.md
Comment (jgerh) [inline docs/guides/overview.md]:
```suggestion
NeMo AutoModel is organized around two key concepts: recipes and components.
```

Comment (jgerh) [inline docs/guides/overview.md:14]:
```suggestion
## Large Language Models (LLM)

```

Comment (jgerh) [inline docs/guides/overview.md:52]:
```suggestion
## Vision Language Models (VLM)

```

File: docs/guides/overview.md [hunk 8]
```diff
@@ -65,6 +65,25 @@ Simple generation script and configurations for VLMs.
 
 - Folder: [examples/vlm_generate](https://github.com/NVIDIA-NeMo/Automodel/tree/main/examples/vlm_generate)
 
+## NeMo Retriever
```
Comment (jgerh) [inline docs/guides/overview.md:70]:
```suggestion
## NeMo Retriever

```

**Review by jgerh — COMMENTED** [review]
Completed tech pubs review of docs/guides/overview.md and docs/guides/retriever/embedding-finetuning.md and provided a few copyedits. docs/index.md reviewed in https://github.com/NVIDIA-NeMo/Automodel/pull/1573.
